src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,38 +1,83 @@
+import { useEffect, useState } from 'react';
+
+//components
+
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  selectedTodo: Todo;
+  setSelectedTodo: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [userData, setUserData] = useState<User>(null);
+  const [hasErrorMessage, setErrorMessage] = useState('');
+
+  const userId = selectedTodo.userId;
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    if (userId) {
+      getUser(userId)
+        .then(data => setUserData(data))
+        .catch(e => setErrorMessage(e))
+        .finally(() => setIsLoading(false));
+    }
+  }, [userId]);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {isLoading ? (
         <Loader />
       ) : (
         <div className="modal-card">
           <header className="modal-card-head">
+            {hasErrorMessage && (
+              <p>An Error occured while downloading user data</p>
```

Comment:
There is a typo in the error message: 'occured' should be 'occurred'. Ensure that error messages are clear and free of typos to improve user experience.